packages/glsp-sprotty/src/ui-extension-override.ts

```diff
@@ -0,0 +1,154 @@
+            return this.containerElement;
+        }
+        // check if the container already exists, independent from any potential parent container
+        // this allows us to use existing elements defined anywhere in the document
+        const existingContainer = this.getContainer();
+        if (existingContainer) {
+            return existingContainer;
+        }
+        // to create a container the parent container
+        const parent = this.getParentContainer();
+        if (!parent || !parent.isConnected) {
+            throw new Error(`Could not obtain attached parent for initializing UI extension ${this.id}`);
+        }
+        const container = this.createContainer(parent);
+        this.insertContainerIntoParent(container, parent);
+        return container;
+    }
+
+    protected getContainer(): HTMLElement | null {
+        return document.querySelector<HTMLElement>(this.containerSelector);
+    }
+
+    protected createContainer(parent: HTMLElement): HTMLElement {
+        const container = document.createElement('div');
+        container.id = parent.id + '_' + this.id();
+        return container;
+    }
+
+    protected initializeContainer(container: HTMLElement): void {
+        container.classList.add(CSS_UI_EXTENSION_CLASS, this.containerClass());
+    }
+
+    protected getParentContainer(): HTMLElement {
+        return document.querySelector<HTMLElement>(this.parentContainerSelector)!;
+    }
+
+    protected insertContainerIntoParent(container: HTMLElement, parent: HTMLElement): void {
+        parent.insertBefore(container, parent.firstChild);
+    }
+
+    protected override setContainerVisible(visible: boolean): void {
+        // the parent class simply sets the style directly, however classes provide more fine-grained control
+        if (visible) {
+            this.containerElement?.classList.remove(CSS_HIDDEN_EXTENSION_CLASS);
+        } else {
+            this.containerElement?.classList.add(CSS_HIDDEN_EXTENSION_CLASS);
+        }
+    }
+
+    protected isContainerVisible(): boolean {
+        return this.containerElement && !this.containerElement.classList.contains(CSS_HIDDEN_EXTENSION_CLASS);
+    }
+
+    protected toggleContainerVisible(): void {
+        this.setContainerVisible(!this.isContainerVisible());
+    }
+}
+
+@injectable()
+export class EditLabelUI extends SprottyEditLabelUI {
+    protected override initializeContents(containerElement: HTMLElement): void {
+        super.initializeContents(containerElement);
+        containerElement.classList.add(CSS_UI_EXTENSION_CLASS);
+    }
+
+    protected override setContainerVisible(visible: boolean): void {
+        if (visible) {
+            this.containerElement?.classList.remove(CSS_HIDDEN_EXTENSION_CLASS);
+        } else {
+            this.containerElement?.classList.add(CSS_HIDDEN_EXTENSION_CLASS);
+        }
+    }
+}
+
+@injectable()
+export class CommandPalette extends SprottyCommandPalette {
```

Comment:
I understand why the customization for the `commandPalette` and `EditLabelUI` are necessary.
However, we should probably contribute the modifications to the `AbstractUIExtension` back to sprotty,
so that we can directly consume it from there in a future release.
This would ensure that the flexible parent/container options can also be used with the UIExtensions reused from sprotty.

Could you please open a ticket to track this?